rotkehlchen/chain/ethereum/modules/balancer/utils.py

```diff
@@ -0,0 +1,284 @@
+    # the IN/OUT tokens but we can't get the name and decimalswhich means
+    # we can get all the name/decimals for the UnknownEthereumToken
+    try:
+        raw_address_tokens = {raw_token['address']: raw_token for raw_token in raw_tokens}
+        raw_token0 = raw_address_tokens[attr_raw_addresses['tokenIn']]
+        raw_token1 = raw_address_tokens[attr_raw_addresses['tokenOut']]
+        token0_name = raw_token0['name']
+        token0_decimals = raw_token0['decimals']
+        token1_name = raw_token1['name']
+        token1_decimals = raw_token1['decimals']
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    token0 = get_ethereum_token(
+        symbol=token0_symbol,
+        ethereum_address=attr_addresses['tokenIn'],
+        name=token0_name,
+        decimals=token0_decimals,
+    )
+    token1 = get_ethereum_token(
+        symbol=token1_symbol,
+        ethereum_address=attr_addresses['tokenOut'],
+        name=token1_name,
+        decimals=token1_decimals,
+    )
+    amm_swap = AMMSwap(
+        tx_hash=tx_hash,
+        log_index=log_index,
+        address=attr_addresses['userAddress'],
+        from_address=attr_addresses['caller'],
+        to_address=attr_addresses['poolAddress'],
+        timestamp=timestamp,
+        location=Location.BALANCER,
+        token0=token0,
+        token1=token1,
+        amount0_in=amount0_in,
+        amount1_in=AssetAmount(ZERO),
+        amount0_out=AssetAmount(ZERO),
+        amount1_out=amount1_out,
+    )
+    return amm_swap
+
+
+def deserialize_token_price(
+        raw_token_price: Dict[str, Any],
+) -> Tuple[ChecksumEthAddress, Price]:
+    """May raise DeserializationError"""
+    try:
+        token_address = raw_token_price['id']
+        usd_price = deserialize_price(raw_token_price['price'])
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    try:
+        token_address = to_checksum_address(token_address)
+    except ValueError as e:
+        raise DeserializationError(
+            f'Invalid ethereum address: {token_address} in token price.',
+        ) from e
+
+    return token_address, usd_price
+
+
+def deserialize_token_day_data(
+        raw_token_day_data: Dict[str, Any],
+) -> Tuple[ChecksumEthAddress, Price]:
+    """May raise DeserializationError"""
+    try:
+        token_address = raw_token_day_data['token']['id']
+        usd_price = deserialize_price(raw_token_day_data['priceUSD'])
+    except KeyError as e:
+        raise DeserializationError(f'Missing key: {str(e)}.') from e
+
+    try:
+        token_address = to_checksum_address(token_address)
+    except ValueError as e:
+        raise DeserializationError(
+            f'Invalid ethereum address: {token_address} in token day data.',
+        ) from e
+
+    return token_address, usd_price
+
+
+def calculate_amm_trade_from_amm_swaps(swaps: List[AMMSwap]) -> AMMTrade:
```

Comment:
this function needs to nevver be called with an empty list, right? So perhaps add an assertion or in the docstring

Reply:
I see, right. "It should never happen" :D. I will add an assertion error as it can only happen during development. The current implementation only triggers the logic of converting swaps to trades if the address has swaps.